### PR TITLE
perf changes

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen-signal

### CODE REVIEW

File: Mantle/MTLModel.h [hunk 1]
```diff
@@ -78,7 +78,7 @@ typedef enum : NSUInteger {
 
 /// Merges the value of the given key on the receiver with the value of the same
 /// key from the given model object, giving precedence to the other model object.
-- (void)mergeValueForKey:(NSString *)key fromModel:(id<MTLModel>)model;
+//- (void)mergeValueForKey:(NSString *)key fromModel:(id<MTLModel>)model;
```
Comment (charlesmchen-signal) [inline Mantle/MTLModel.h:87]: Should these two changes have `BEGIN ORM-PERF-...` markers?

File: Mantle/MTLModel.m [hunk 2]
```diff
@@ -215,11 +242,28 @@ + (NSSet *)permanentPropertyKeys {
...
+        dictionaryValueKeys = keys.allObjects;
+        objc_setAssociatedObject(self, MTLModelCachedDictionaryValueKeysKey, dictionaryValueKeys, OBJC_ASSOCIATION_COPY);
+    }
+    return dictionaryValueKeys;
+}
 
-	return [self dictionaryWithValuesForKeys:keys.allObjects];
+- (NSDictionary *)dictionaryValue {
+    NSArray<NSString *> *keys = self.class.dictionaryValueKeys;
+	return [self dictionaryWithValuesForKeys:keys];
```
Comment (charlesmchen-signal) [inline Mantle/MTLModel.m:265]: Weird indent?

File: Mantle/MTLReflection.m [hunk 3]
```diff
@@ -24,27 +24,34 @@ SEL MTLSelectorWithKeyPattern(NSString *key, const char *suffix) {
...
-
-	memcpy(selector + prefixLength + initialLength + restLength, suffix, suffixLength);
-	selector[prefixLength + initialLength + restLength + suffixLength] = '\0';
-
-	return sel_registerName(selector);
-}
+// BEGIN ORM-PERF-1
+// Commented out by mkirk as part of ORM perf optimizations.
+// The `MTLSelectorWithCapitalizedKeyPattern` can be quite expensive in aggregate
+// and we're aren't using the reflective features that require it.
```
Comment (charlesmchen-signal) [inline Mantle/MTLReflection.m]: s/and we're aren't/and we aren't/

**Review by charlesmchen-signal — APPROVED** [review]
LGTM

File: Mantle/MTLModel.m (hunk 2 above)
Comment (michaelkirk-signal) [reply Mantle/MTLModel.m:265]: Oh wow, tabs.

File: Mantle/MTLReflection.m (hunk 3 above)
Comment (michaelkirk-signal) [reply Mantle/MTLReflection.m]: Fixed here and elsewhere

File: Mantle/MTLModel.h (hunk 1 above)
Comment (michaelkirk-signal) [reply Mantle/MTLModel.h:87]: Added, thanks.

**Pull request merged**